### DISCUSSION TITLE
ZIndex of component

### DISCUSSION THREAD

**Discussion by a313** [flame-engine/flame#902] [Q&A] [answered]
Hi everyone, 
I have a Draggable component and i want to change z index of that component when dragging. But I didnt found any solution for this. Any suggest?

**Comment by rundun** [comment, marked as answer]
There is a priority property on components that you potentially could leverage off of.

**Reply by a313** [reply]
Thanks for your reply. I checked but priority has only get method. I cant change it realtime

**Reply by rundun** [reply]
You need to override it.

@override
int get priority => yourNumber;

**Reply by spydon** [reply]
You can change it real-time. You'll have to use the `BaseGame.changePriority` method, since it will have to re-balance the component tree.

**Reply by a313** [reply]
I have try something like that in draggable component
  @override
  int get priority => isDragging ? 999 : 0;

And in BaseGame

  if (component is DraggableComponent && component.isDragging) {
      return super.changePriority(component, 999, reorderRoot: true);
    } else {
      return super.changePriority(component, 0, reorderRoot: false);
    }
    
But it not work at all. Can you show me some example. Thanks

**Reply by rundun** [reply]
Paste a minimal code example off your problem

**Reply by spydon** [reply]
@a313 you can't change the priority dynamically like that without re-ordering the component tree, so instead you should override `onDragStart` and put `gameRef.changePriority` in there. (Add the `hasGameRef` mixin to your component so that you have access to the `gameRef`).
Have a look at the code for the priority example:
https://github.com/flame-engine/flame/blob/main/examples/lib/stories/components/priority.dart
https://examples.flame-engine.org/#/Components_Priority

**Reply by a313** [reply]
> @a313 you can't change the priority dynamically like that without re-ordering the component tree, so instead you should override `onDragStart` and put `gameRef.changePriority` in there. (Add the `hasGameRef` mixin to your component so that you have access to the `gameRef`).
> Have a look at the code for the priority example:
> https://github.com/flame-engine/flame/blob/main/examples/lib/stories/components/priority.dart
> https://examples.flame-engine.org/#/Components_Priority

Thank you!
